worker/pkg/benthos/sql/output_sql_insert.go

```diff
@@ -255,28 +268,86 @@ func (s *pooledInsertOutput) WriteBatch(ctx context.Context, batch service.Messa
 			return fmt.Errorf("mapping returned non-array result: %T", iargs)
 		}
 
-		// set any default transformations
-		for idx, a := range args {
-			if a == "DEFAULT" {
-				args[idx] = goqu.L("DEFAULT")
+		rows = append(rows, args)
+	}
+
+	filteredCols, filteredRows := filterIdentityColumns(s.driver, s.identityColumns, s.columns, rows)
+
+	// set any default transformations
+	for i, row := range filteredRows {
+		for j, arg := range row {
+			if arg == "DEFAULT" {
+				filteredRows[i][j] = goqu.L("DEFAULT")
 			}
 		}
-
-		rows = append(rows, args)
 	}
 
-	insertQuery, err := querybuilder.BuildInsertQuery(s.driver, fmt.Sprintf("%s.%s", s.schema, s.table), s.columns, rows, &s.onConflictDoNothing)
+	insertQuery, err := querybuilder.BuildInsertQuery(s.driver, fmt.Sprintf("%s.%s", s.schema, s.table), filteredCols, filteredRows, &s.onConflictDoNothing)
 	if err != nil {
 		return err
 	}
 
+	if s.driver == sqlmanager_shared.MssqlDriver && len(filteredCols) == 0 {
+		insertQuery = getMssqlDefaultValuesInsertSql(s.schema, s.table, len(rows))
+	}
+
 	query := s.buildQuery(insertQuery)
 	if _, err := s.db.ExecContext(ctx, query); err != nil {
 		return err
 	}
 	return nil
 }
 
+// use when all columns are identity generation columns
+func getMssqlDefaultValuesInsertSql(schema, table string, rowCount int) string {
+	var sql string
+	for i := 0; i < rowCount; i++ {
+		sql += fmt.Sprintf("INSERT INTO %s.%s DEFAULT VALUES;", schema, table)
+	}
+	return sql
+}
+
+func filterIdentityColumns(
```

Comment:
nit: Is this thing returning cols without identity-specific columns? Kinda what it looks like from the code but not immediately clear from the func name.

Also not obvious that it only seems to do so if the arg is DEFAULT. But maybe it is always that? 
Maybe a func comment could be helpful here.